crates/admin/src/cluster_controller/service.rs

```diff
@@ -268,7 +285,20 @@ impl<T: TransportConnect> Service<T> {
                     }
                 }
                 Ok(cluster_state) = cluster_state_watcher.next_cluster_state() => {
-                    scheduler.on_cluster_state_update(cluster_state).await?;
+                    observed_cluster_state.update(&cluster_state);
+                    logs_controller.on_observed_cluster_state_update(&observed_cluster_state)?;
+                    scheduler.on_observed_cluster_state(&observed_cluster_state).await?;
+                }
+                result = logs_controller.run_async_operations() => {
+                    result?;
```

Comment:
[⬇️ Just asking] Do we have to be extra careful not to crash the controller task here and elsewhere? Would an error result bring down the entire server, or just stop the task? If the latter, we should possibly have a supervisor to restart it - or alternatively make sure it crash-stops the entire process.

Reply:
Ack, thanks for confirming!